### ISSUE TITLE
Add unedited line for diff previews that only have either adds or delete

### ISSUE THREAD

**Issue by ematipico** [romefrontend/rome#679]
## First issue

I noticed few things inside the rule `noUselessFragments`. The test assumes that the following is invalid:

```js
<>
  <div>
  <div>
</>
```

Isn't it wrong? In this case the fragment is needed, am I right? 

Sure, it depends on the case:

```js
// not needed
<section>
	<>
	  <div>
	  <div>
	</>
<section/>

// needed
return (
	<>
	  <div>
	  <div>
	</>
)

```

## Second issue

It's not technically an issue but I think that the suggested fix could be improved. This is a piece of snapshot taken from the source code:

```js
unknown:2:5 lint/react/noUselessFragment  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Avoid using unnecessary Fragment.

    1 │ <section>
  > 2 │      <>
      │      ^^
  > 3 │       <div />
  > 4 │       <div />
  > 5 │      </>
      │ ^^^^^^^^
    6 │     </section>
    7 │     

  ℹ A Fragment is redundant if it contains only one child, or if it is the child of a html element,
    and is not a keyed fragment.

  ℹ Recommended fix

    1  │ - <>
    2  │ -             <div />
    3  │ -             <div />
    4  │ -           </>
```

The suggested fix is to remove everything; I think we can improve the suggested fix by just removing the fragment and keeping what's inside it

**Comment by sebmck** [comment]
> The suggested fix is to remove everything; I think we can improve the suggested fix by just removing the fragment and keeping what's inside it

I think it might be a bug in the way we're displaying the "Recommend fix" diff. Lines 2-3 probably only have whitespace removed. Could you run `check --apply` on a file and see what it's fixed to?

**Comment by ematipico** [comment]
Yeah you were right, it's just presentation of the that is misleading. I just tried the following code and run `check --apply`:

```
// before
export function App() {
    return (
        <section>
            <>
                <div></div>
                <div></div>
                <div></div>
            </>
        </section>
    )
}

// after
export function App() {
	return <section>
            
                <div></div>
                <div></div>
                <div></div>
            
        </section>;
}
```


Instead, here's what I get when I run `check`:

<img width="1023" alt="Screenshot 2020-07-02 at 08 17 52" src="https://user-images.githubusercontent.com/602478/86328352-97db1b00-bc3c-11ea-8150-96e8125a4cd6.png">

And this the pending fixes:

<img width="772" alt="Screenshot 2020-07-02 at 08 18 02" src="https://user-images.githubusercontent.com/602478/86328393-a7f2fa80-bc3c-11ea-8766-99fcece642d7.png">

**Comment by sebmck** [comment]
Looks like it's just an unhandled case in the way we output diffs. When we have a lint that only contains deletions then we should output the final as a separate line. I think we also need a better way to indicate tabs. Previously we used a special character but with the switch to tabs we started treating a hard tab as two spaces.

**Comment by sebmck** [comment]
I'll leave this issue open (and wont mark it as done from my notifications) until I can fix it. The diffing code is pretty gnarly.

**Comment by sebmck** [comment]
Fixed in #739 where I rewrote our unified diff algorithm. Before I didn't really know what I was doing but I'm extremely confident now that we have parity with GitHub and other source code tools. I added your example as a test case:

```
# Before
<section>
	<>
		<div />
		<div />
	</>
</section/>

# After
<section>
	<div />
	<div />
</section>

# Diff
  1 1 │   <section>
  2   │ -   <>
  3   │ -     <div />
  4   │ -     <div />
  5   │ -   </>
  6   │ - </section/>
    2 │ +   <div />
    3 │ +   <div />
    4 │ + </section>
```

Should be pretty easy to read since the same reading rules apply if you're used regular source control diffs.

Thank you for the report @ematipico!
